gate-oauth2/src/main/groovy/com/netflix/spinnaker/gate/security/oauth2/SpinnakerUserInfoTokenServices.groovy

```diff
@@ -96,14 +107,39 @@ class SpinnakerUserInfoTokenServices implements ResourceServerTokenServices {
     }
 
     def username = details[userInfoMapping.username] as String
-    def roles = getRoles(details)
+    def roles = getRoles(details) ?: []
 
     // Service accounts are already logged in.
     if (!isServiceAccount) {
-      if (roles.isEmpty()) {
-        permissionService.login(username)
-      } else {
-        permissionService.loginWithRoles(username, roles)
+      def id = registry
```

Comment:
heavily inspired by [gate-saml](https://github.com/spinnaker/gate/blob/2d70a501ad70fafafd5c3674183f1470a2c5f821/gate-saml/src/main/groovy/com/netflix/spinnaker/gate/security/saml/SamlSsoConfig.groovy#L216)

I didn't add support for specifying requiredRoles because that can be accomplished via the userInfoRequirements configuration (and also via config at the IdP..)